src/shader.cpp

```diff
@@ -374,14 +401,6 @@ void ShaderProgram::setUniforms()
     }
 }
 
-
-void ShaderProgram::setContext(const QGLContext* context)
-{
-    m_context = context;
-    setShader(shaderSource());
-}
```

Comment:
I admit I had some problems keeping the context around as `m_context`, and I was finding odd things happening.  I can see it's removed here - did you find anything enlightening about what was going wrong?